astropy/modeling/_fitting_parallel.py

```diff
@@ -127,7 +127,7 @@ def _fit_models_to_chunk(
     for index in np.ndindex(iterating_shape_chunk):
         # If all data values are NaN, just set parameters to NaN and move on
         if np.all(np.isnan(data[index])):
-            for ipar, name in enumerate(model.param_names):
+            for ipar, _ in enumerate(model.param_names):
```

Comment:
If only the index is used this can be rewritten as 
```suggestion
            for ipar in range(len(model.param_names)):
```

Reply:
And similarly for some of the other changes.